Dangerfile

```diff
@@ -215,97 +252,103 @@ if @platform == "ios"
+
+# Make it more obvious that a PR is a work in progress and shouldn't be merged yet
+warn("PR is classed as Work in Progress") if github.pr_title.include? "[WIP]"
+
+# Warn when there is a big PR
+warn("Big PR") if git.lines_of_code > 500
+
+# Mainly to encourage writing up some reasoning about the PR, rather than
+# just leaving a title
+if github.pr_body.length < 5
+  fail "Please provide a summary in the Pull Request description"
 end
 
-########################
-#    Android SECTION   #
-########################
-if @platform == "android"
+# Common files
+files_to_check = ["Gemfile.lock", ".travis.yml", ".gitignore", "Dangerfile"]
+# Node files
+files_to_check += ["yarn.lock", "docker-compose.yml", "Procfile", "npm-shrinkwrap.json", "node_modules", "tasks/options/env.coffee", "tslint.json"]
+# iOS files
+files_to_check += ["Cakefile", "fastlane/settings.yml.erb", "fastlane/Fastfile", "Podfile.lock"]
+# Check if files were modified
+(modified_files & files_to_check).each do |file|
+  message("`#{file}` modified")
+end
 
-  modified_files.each do |file|
-    ext = File.extname(file)
-    case ext
-    # Warn when a file .gradle is modified
-    when ".gradle"
-      message("`#{file}` was modified")
-    end
-    # Warn when a FileManifest.xml is modified
-    message("`#{file}` was modified") if file =~ /Manifest\.xml/
+
+# Warn if 'Gemfile' was modified and 'Gemfile.lock' was not
+if modified_files.include?("Gemfile")
+  if !modified_files.include?("Gemfile.lock")
+    warn("`Gemfile` was modified but `Gemfile.lock` was not")
+  else
+    message("`Gemfile` modified")
   end
 end
 
-########################
-#      Web SECTION     #
-########################
-if @platform == "web"
-  modified_files.each do |file|
-    begin
-      checkForEnginesVersion(file)
-      checkForNpmInstallGlobal(file)
-    rescue
-      message "Could not read file #{file}, does it really exist?"
-    end
+if @platform == "ios"
+  validatePodfile(modified_files)
+  checkCakefileMissconfig("Cakefile") if modified_files.include?("Cakefile")
+  validatePlistFiles(modified_files)
+elsif @platform == "nodejs"
+  # Warn if 'console.log' was added
+  diff = github.pr_diff
+  warn("`console.log` added") if diff =~ /\+\s*console\.log/
+end
 
-    ext = File.extname(file)
-    case ext
-    # Warn when a file .style is modified
-    when ".styl"
-      message("`#{file}` was modified")
-    when ".ts"
-      begin
-        File.foreach(file) do |line|
-          line = line.gsub('\n','').strip
-
-          checkForWeaklyTypedFunctionReturn(line)
-        end
-      rescue
-        message "Could not read file #{file}, does it really exist?"
-      end
-    end
-  end
+validatePackageJson(modified_files, github.pr_diff) if @platform == "nodejs" || @platform == "web"
 
-  validatePackageJson(modified_files, github.pr_diff)
+modified_files.each do |file|
+  begin
+    checkForFile(file)
+  rescue
+    message "Could not read file #{file}, does it really exist?"
```

Comment:
Can we check if file was excluded or moved before raising this error?

Most of my PRs have a long list with this kind of message that is just noise and can be completely ignored by devs. example: https://github.com/indigotech/br-pro120-jogo_das_forcas-server-api/pull/25#issuecomment-290737206